httomo/runner/task_runner.py

```diff
@@ -290,12 +292,20 @@ def _count_tuple_values(self, d: Dict[str, Any]) -> int:
 
     def determine_max_slices(self, section: Section, slicing_dim: int):
         assert self.source is not None
-        data_shape = self.source.chunk_shape
+        assert len(section) > 0, "Section should contain at least 1 method"
 
+        data_shape = self.source.chunk_shape
         max_slices = data_shape[slicing_dim]
-        if len(section) == 0:
-            section.max_slices = min(httomo.globals.MAX_CPU_SLICES, max_slices)
-            return
+
+        # loop over all methods in section
+        has_gpu = False
+        for idx, m in enumerate(section):
+            if m.implementation in ["gpu", "gpu_cupy"]:
+                has_gpu = True
+
+        # if section consists of all cpu method then MAX_CPU_SLICES defines the block size
+        if not has_gpu:
+            section.max_slices = min(MAX_CPU_SLICES, max_slices)
```

Comment:
If this condition is encountered (ie, we have confirmed that all methods in the section are CPU), is there any need to carry on with the rest of the code in this function? Can we just `return` after taking the max slices to be the min of the `MAX_CPU_SLICES` and `max_slices`?

Reply:
good point, it also checked for `m.is_gpu` bellow so I added it in the condition,  in order to remove the bit bellow 
`if m.implementation in ["gpu", "gpu_cupy"] or m.is_gpu:`